man/landscape-config.txt
NAME
  landscape-config - configure the Landscape management client

SYNOPSIS
  landscape-config [options]

DESCRIPTION

Before using the landscape-client it must be configured with account and
computer information that is transmitted to the Landscape server when the
client connects. The registration can be performed with or without user
interaction.

Running the landscape-config program without arguments will start the
program, prompting you for the necessary information to run the client.
Optionally you can provide command-line arguments (detailed below) to
specify default responses.

You will be told if the registration was successful or if an error occurred.
When registration is successful Landscape is configured and running on your
system. Errors can occur if you provide incorrect registration details or if
there are network issues. In the latter case, the client will keep trying
to complete the registration in the background.

OPTIONS
  --version                Show program's version number and exit.
  -h, --help               Show this help message and exit.
  -c FILE, --config=FILE   Use config from this file (any command line options
                           override settings from the file) (default: '/etc/landscape/client.conf').
  -d PATH, --data-path=PATH   The directory to store data files in (default: '/var/lib/landscape/client').
  -q, --quiet              Do not log to the standard output.
  -l FILE, --log-dir=FILE   The directory to write log files to (default: '/var/log/landscape').
  --log-level=LOG_LEVEL    One of 'debug', 'info', 'warning', 'error' or 'critical' (default: 'info').
  --ignore-sigint          Ignore interrupt signals.
  --ignore-sigusr1         Ignore SIGUSR1 signal to rotate logs.
  -a NAME, --account-name=NAME   The account this computer belongs to.
  -p KEY, --registration-key=KEY  The account-wide key
                           used for registering clients.
  -t TITLE, --computer-title=TITLE   The title of this computer.
  -u URL, --url=URL        The server URL to connect to (default: 'https://landscape.canonical.com/message-system').
  -k SSL_PUBLIC_KEY, --ssl-public-key=SSL_PUBLIC_KEY   The public SSL key is 
                             a CA Certificate to verify the server. Only 
                             used if the given server URL is https.
  --exchange-interval=INTERVAL   The number of seconds between server
                           exchanges (default: 900, min: 60, max: 86400).
  --urgent-exchange-interval=INTERVAL   The number of seconds between urgent
                           server exchanges (default: 60, min: 10, max: 3600).
  --ping-interval=INTERVAL  The number of seconds between pings (default: 30,
                           min: 10, max: 3600).
  --ping-url=PING_URL      The URL to perform lightweight exchange initiation
                           with (default: 'http://landscape.canonical.com/ping').
  --package-monitor-interval=PACKAGE_MONITOR_INTERVAL  The interval between
                           package monitor runs (default: 1800, min: 60,
                           max: 86400).
  --snap-monitor-interval=SNAP_MONITOR_INTERVAL  The interval between snap
                           monitor runs (default: 1800, min: 60, max: 86400).
  --apt-update-interval=APT_UPDATE_INTERVAL  The interval between apt update
                           runs (default: 21600, min: 600, max: 604800).
  --apt-update-timeout=APT_UPDATE_TIMEOUT  The maximum time in seconds to wait
                           for apt update to complete before killing it
                           (default: 1800).
  --flush-interval=INTERVAL   The number of seconds between flushes to disk for
                           persistent data (default: 300, min: 10,
                           max: 3600).
  --stagger-launch=STAGGER_RATIO  Ratio, between 0 and 1, by which to stagger various
                           tasks of landscape.
  --http-proxy=URL         The URL of the HTTP proxy, if one is needed.
  --https-proxy=URL        The URL of the HTTPS proxy, if one is needed.
  --access-group           Suggested access group for this computer.
  --tags=TAGS              Comma separated list of tag names to be sent to the
                           server.
  --import=FILENAME_OR_URL   Filename or URL to import configuration from.
                           Imported options behave as if they were passed in
                           the command line, with precedence being given to
                           real command line options.
  --script-users=USERS     A comma-separated list of users to allow scripts to
                           run. To allow scripts to be run by any user,
                           enter: ALL.
  --script-tempdir=SCRIPT_TEMPDIR  The working directory to use for script executions.
                           Must have read, write, and exec privileges for
                           any script users
  --include-manager-plugins=PLUGINS   A comma-separated list of manager
                           plugins to load explicitly.
  --manage-sources-list-d [MANAGE_SOURCES_LIST_D]  Repository profiles manage
                           the files in ’etc/apt/sources.list.d'. (default: true)
  -n, --no-start           Don't start the client automatically.
  --ok-no-register         Return exit code 0 instead of 2 if the client can't be
                           registered.
  --silent                 Run without manual interaction.
  --disable                Stop running clients and disable start at boot.
  --init                   Set up the client directories structure and exit.
  --is-registered          (Deprecated) Same as --registration-sent. Use 
         --registration-sent instead. This option will be removed in the 26.04 release.
         Exit with code 0 (success) if client
			   has sent registration request else returns 5.
			   Display registration sent info.
			   (NOTE: use --actively-registered to detect if
			   registration has been accepted server-side)
  --registration-sent      Exit with code 0 (success) if client
			   has sent registration request else returns 5.
			   Display registration sent info.
			   (NOTE: use --actively-registered to detect if
			   registration has been accepted server-side)
  --actively-registered    Exit with code 0 (success) if client
                           is registered else returns 5. Display
                           registration info.
  --show                   Output all configuration data as plain text and exit.
  --show-json              Output all configuration data as JSON and exit.

CLOUD

Landscape has some cloud features that become available when the EC2 or
UEC machine instance was started using Landscape and the AMI is one of
the official ones provided in the Web user interface. We call these
instances "Landscape enabled", because they contain a pre-configured
landscape-client installed in them which will register the running instance
automatically with Landscape as soon as it starts up.

You can deploy your own AMI, but if you wish the instance to become
"Landscape managed" you need to take a few steps:
* make sure the cloud is created in Landscape
* add "CLOUD=1" to /etc/default/landscape-client
* make sure the client is configured to start at boot (i.e., the
/etc/default/landscape-client has the line "RUN=1")

There is no need to further configure the /etc/landscape/client.conf file
with details such as account or key, because when in cloud mode
this is all discovered by the client itself.

You can avoid this all if you just re-bundle the AMIs we provide.
landscape-client is already configured and prepared for the cloud in
them.

EXAMPLES

Register a machine for the first time, or reconfigure an already
registered machine, interactively. Command line parameters suppress
interaction for provided values.

  landscape-config

Register a machine for the first time, or reconfigure an already
registered machine, without requiring user interaction. The client
will be configured to start on boot automatically:

  landscape-config --silent -a account-name -p secret -t `hostname`

Register a machine with the script execution plugin enabled, without
requiring user interaction:

  landscape-config --silent -a account-name -p secret -t `hostname` --script-users nobody,landscape,root

Register a machine with some tags:

  landscape-config --silent -a account-name -p secret -t `hostname` --tags=server,www

To disable a client, stopping current instances and disabling start at
boot:

  landscape-config --disable


SEE ALSO

  landscape-client (1)

AUTHOR
  Landscape Development Team <landscape-team@canonical.com>


debian/changelog
landscape-client (26.08.1-0landscape0) jammy; urgency=medium

  * fix: broker stale socket recovery
  * fix: client config file permissions in snap (LP: #2161005)

 -- David McLain <david.mclain@canonical.com>  Fri, 28 Aug 2026 17:15:39 -0500

landscape-client (26.08-0landscape0) jammy; urgency=medium

  * refactor: cap error count for exponential backoff

 -- Joey Mucci <joseph.mucci@canonical.com>  Wed, 12 Aug 2026 11:10:43 -0400

landscape-client (26.08~beta.4-0landscape0) jammy; urgency=medium

  * fix: set User-Agent for ping requests
  * fix: disconnect connectors on error paths to prevent socket leak (LP: #2147687)

 -- Joey Mucci <joseph.mucci@canonical.com>  Fri, 22 May 2026 16:44:53 -0400

landscape-client (26.08~beta.3-0landscape0) jammy; urgency=medium

  * fix(snap): make url options consistent with landscape-config
  * feat: add configurable timeout for apt-update (LP: #2131251)

 -- Joey Mucci <joseph.mucci@canonical.com>  Fri, 24 Apr 2026 18:11:56 -0400

landscape-client (26.08~beta.2-0landscape0) jammy; urgency=medium

  * fix: snap builds & installation

 -- Joey Mucci <joseph.mucci@canonical.com>  Fri, 10 Apr 2026 18:00:56 -0400

landscape-client (26.08~beta.1-0landscape0) jammy; urgency=medium

  * fix: correct string formatting for ubuntu release upgrader log
  * fix: restore tests for amd64v3
  * fix: data file permissions are too lax (LP: #2121558)
  * fix: add deterministic sub-process generation for flaky tests

 -- Joey Mucci <joseph.mucci@canonical.com>  Tue, 07 Apr 2026 18:28:24 -0400

landscape-client (26.02.1-0landscape0) jammy; urgency=medium

  * fix: restore functionality and tests for python 3.14

 -- Joey Mucci <joseph.mucci@canonical.com>  Thu, 05 Feb 2026 17:38:31 +0000

landscape-client (26.02-0landscape0) jammy; urgency=medium

  * feat: add FDE recovery key manager plugin
  * fix: package changer uses proxy when configured

 -- Joey Mucci <joseph.mucci@canonical.com>  Mon, 02 Feb 2026 15:57:02 +0000

landscape-client (25.10.1-0landscape0) jammy; urgency=medium

  * refactor: Revert addition of --ssl-ca flag
  * fix: Move WSL config options out of unsaved_options
  * build: add ruff to dev-lxc config
  * fix: landscape-config reads deprecated ssl_public_key field
  * refactor: clean up bpickle dumps
  * build: use ruff for formatting and linting

 -- Jacob Chin <jacob.chin@canonical.com>  Tue, 25 Nov 2025 16:27:53 -0500

landscape-client (25.10-0landscape0) jammy; urgency=medium

  * build: update snap-http module
  * build: release 25.10

 -- Spencer Runde <spencer.runde@canonical.com>  Wed, 29 Oct 2025 12:59:24 -0500

landscape-client (25.10~rc3-0landscape0) jammy; urgency=medium

  * fix: don't explicitly make root owner of the executables

 -- Ardavan Behnia <ardavan.behnia@canonical.com>  Thu, 23 Oct 2025 15:29:06 -0700

landscape-client (25.10~rc2-0landscape0) jammy; urgency=medium

  * fix: add python version variable for setup.py

 -- Ardavan Behnia <ardavan.behnia@canonical.com>  Fri, 17 Oct 2025 13:02:59 -0700

landscape-client (25.10~rc1-0landscape0) jammy; urgency=medium

  * feat: add activity to detach pro
  * feat: add activity for attaching pro
  * feat: add --show argument to landscape-config
  * fix: example.conf pro management plugin
  * fix: add id to devmode snaps for server indexing
  * fix: add python3-packaging dependency to landscape-client
  * fix: uaclient wrapper with snap and core devices
  * fix: make improvements to landscape-config --show

 -- Ardavan Behnia <ardavan.behnia@canonical.com>  Thu, 16 Oct 2025 12:32:27 -0700

landscape-client (25.08.2+git6481-0landscape0) jammy; urgency=medium

  * fix: add deprecation warning for --is-registered option for landscape-config
  * feat: add --show and --show-json options for landscape-config
  * feat: add activity for attaching a pro token
  * feat: add activity for detaching a pro token
  * fix: bug in uaclient wrapper on core devices
  * fix: add explicit dependency for python3-packaging
  * fix: add id for devmode snaps for server indexing (LP: #2121798)

 -- Bill Kronholm <bill.kronholm@canonical.com>  Fri, 10 Oct 2025 13:33:22 -0700

landscape-client (25.08.2+git6468-0landscape0) noble; urgency=medium

  * feat: rename ssl-public-key to ssl-ca. Accept either.

 -- Spencer Runde <spencer.runde@canonical.com>  Fri, 29 Aug 2025 15:56:49 -0500

landscape-client (25.08.2-0landscape0) jammy; urgency=medium

  * build: added packaging to build-depends

 -- David McLain <david.mclain@canonical.com>  Thu, 14 Aug 2025 23:44:58 +0000

landscape-client (25.08.1-0landscape0) jammy; urgency=medium

  * build: added setuptools to build-depends
  * fix: setup.py clean command

 -- David McLain <david.mclain@canonical.com>  Thu, 14 Aug 2025 17:13:13 -0500

landscape-client (25.08-0landscape0) jammy; urgency=medium

  * feat: add --script-tempdir configuration

 -- David McLain <david.mclain@canonical.com>  Thu, 14 Aug 2025 15:39:02 +0000

landscape-client (25.04+git6541-0landscape0) jammy; urgency=medium

  * feat: log pending message count
  * feat: unknown hashes per request is configurable
  * refactor: delete unused message schemas
  * feat: add machine id fields to registration and computer info messages

 -- Bill Kronholm <bill.kronholm@canonical.com>  Wed, 16 Jul 2025 15:56:00 -0700

landscape-client (25.04+git6435-0landscape0) jammy; urgency=medium

  * feat: add optional --authenticated-attach-code to configuration parameters
  * feat: add usgmanager plugin to example.conf
  * feat: configurably exclude package sources
  * fix: remove deprecated use of unittest.makeSuite to get tests passing on
    >=py312
  * fix: make config tests insensitive to http(s) proxy settings
  * fix: add run-id and operation-id fields to usg-audit message
  * fix: ensure usgmanager plugin runs in deferred
  * fix: usgmanager should accept only the profile or the tailoring file
  * fix: .list and .sources files are restored when a repository profile is

landscape/client/broker/config.py
"""Configuration class for the broker."""

import os

from landscape.client.deployment import Configuration


class BrokerConfiguration(Configuration):
    """Specialized configuration for the Landscape Broker.

    @cvar required_options: C{["url"]}
    """

    def __init__(self):
        super().__init__()
        self._original_http_proxy = os.environ.get("http_proxy")
        self._original_https_proxy = os.environ.get("https_proxy")

    @property
    def exchange_store_path(self):
        return os.path.join(self.data_path, "exchange.database")

    def make_parser(self):
        """Parser factory for broker-specific options.

        @return: An L{ArgumentParser} preset for all the options
            from L{Configuration.make_parser} plus:
              - C{account_name}
              - C{registration_key}
              - C{computer_title}
              - C{exchange_interval} (C{15*60})
              - C{urgent_exchange_interval} (C{1*60})
              - C{ping_interval} (C{30})
              - C{http_proxy}
              - C{https_proxy}
              - C{hostagent_uid}
        """
        parser = super().make_parser()

        parser.add_argument(
            "-a",
            "--account-name",
            metavar="NAME",
            help="The account this computer belongs to.",
        )
        parser.add_argument(
            "-p",
            "--registration-key",
            metavar="KEY",
            help="The account-wide key used for registering clients.",
        )
        parser.add_argument(
            "-t",
            "--computer-title",
            metavar="TITLE",
            help="The title of this computer",
        )
        parser.add_argument(
            "--exchange-interval",
            default=15 * 60,
            type=int,
            metavar="INTERVAL",
            help=self.interval_help(
                "exchange_interval",
                "The number of seconds between server exchanges",
                15 * 60,
            ),
        )
        parser.add_argument(
            "--urgent-exchange-interval",
            default=1 * 60,
            type=int,
            metavar="INTERVAL",
            help=self.interval_help(
                "urgent_exchange_interval",
                "The number of seconds between urgent server exchanges",
                1 * 60,
            ),
        )
        parser.add_argument(
            "--ping-interval",
            default=30,
            type=int,
            metavar="INTERVAL",
            help=self.interval_help(
                "ping_interval",
                "The number of seconds between pings",
                30,
            ),
        )
        parser.add_argument(
            "--http-proxy",
            metavar="URL",
            help="The URL of the HTTP proxy, if one is needed.",
        )
        parser.add_argument(
            "--https-proxy",
            metavar="URL",
            help="The URL of the HTTPS proxy, if one is needed.",
        )
        parser.add_argument(
            "--access-group",
            default="",
            help="Suggested access group for this computer.",
        )
        parser.add_argument(
            "--tags",
            help="Comma separated list of tag names to be sent to the server.",
        )
        parser.add_argument(
            "--hostagent-uid",
            help="Only set this value if this computer is a WSL instance "
            "managed by Landscape, in which case set it to be the uid that "
            "Landscape assigned to the host machine.",
        )
        parser.add_argument(
            "--installation-request-id",
            help="Only set this value if this computer is a child instance "
            "managed by Landscape, in which case set it to be the request id "
            "that Landscape assigned to the installation activity for the "
            "host machine.",
        )
        parser.add_argument(
            "--authenticated-attach-code",
            help="A one-time-use, Landscape server-generated code.",
        )
        parser.add_argument(
            "--http-client",
            default="pycurl",
            choices=["pycurl", "urllib"],
            help="The HTTP client to use for server communication (pycurl or urllib).",
        )

        return parser

    @property
    def message_store_path(self):
        """Get the path to the message store."""
        return os.path.join(self.data_path, "messages")

    def load(self, args):
        """
        Load options from command line arguments and a config file.

        Load the configuration with L{Configuration.load}, and then set
        C{http_proxy} and C{https_proxy} environment variables based on
        that config data.
        """
        super().load(args)
        if self.http_proxy:
            os.environ["http_proxy"] = self.http_proxy
        elif self._original_http_proxy:
            os.environ["http_proxy"] = self._original_http_proxy

        if self.https_proxy:
            os.environ["https_proxy"] = self.https_proxy
        elif self._original_https_proxy:
            os.environ["https_proxy"] = self._original_https_proxy
